ci/bash-lib.yml

```diff
@@ -49,5 +49,54 @@ steps:
         trap - EXIT
         eval "$restore_trap"
     }
+    gpg_verify() {
+        local key gpg_dir signature_file res
+        signature_file=$1
+        key=$(mktemp)
+        cat > $key <<PUB_KEY
+    -----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
I don’t know the expiry but probably worth checking to see how soon we have to figure out how to handle this.

Reply:
My ~understanding~ current guess is that the signature has been made with this key and includes a timestamp, so in a way it will never expire: artifacts signed with this key will always need to be verified with this key, regardless of the expiration status of the key. What would trigger a failure is if the signature timestamp is past the key expiration time.

So my current expectation is that when the key expires, we just add a second one, and then we check all artifacts using a keyring that contains both keys.

Reply:
Right, I was more thinking of new artifacts that need a new key rather than this key becoming invalid for old artifacts.

Reply:
Do you foresee any issue with just adding a second `gpg --import` at that point?

Reply:
Sounds sensible, my gpg foo is very weak. I was mostly just seeing a hardcoded key and wondered what would happen once it expires.

Reply:
Ah, well, you'll be seeing two hardcoded keys. 🤷

I mean, obviously at some point this could get out of hands if we produce many keys, but hopefully each one lasts for a few years.